R/dist_spec.R

```diff
@@ -406,9 +406,10 @@ max.dist_spec <- function(x, ...) {
 #' (i.e. those with finite support and constant parameters).
 #' @title Discretise a <dist_spec>
 #' @param x A `<dist_spec>`
-#' @param silent Logical; if `TRUE` then any distribution that can't be
-#'   discretised will be returned as is. If `FALSE` then an error will be
-#'   thrown.
+#' @param strict Logical; If `TRUE` (default) an error will be thrown if a
+#' distribution cannot be discretised (e.g., because no finite maximum has been
+#' specified or parameters are uncertain). If if `FALSE` then any distribution
+#' that cannot be discretised will be returned as is.
 #' @return A `<dist_spec>` where all distributions with constant parameters are
```

Comment:
Do we actually need this argument? `strict` only controls whether an error should be thrown or not. It seems more idiomatic to just error if something doesn't work.

Reply:
My original intention was to remove but then that would basically require reproducing all the code when doing https://github.com/epiforecasts/EpiNow2/blob/4d975edf6098c64f8dbba98bcf4ffb52e29de399/R/create.R#L731

The complication is that we can have composite delay distributions (e.g. the sum of the incubation period and reporting delay), and we might be able to discretise one but not the other. Perhaps the whole thing is overengineered anyway but for now I think we want to give the user the chance to say: discretise whatever you can, and feel free to leave the rest alone.